### PR TITLE
Wire up Azure for v4 resources and add provider to the chartconfig resource

### PR DESCRIPTION
**Pull request by rossf7**
This PR wires up the v4 resources for Azure. It also adds Provider to the chartconfig resource. This is so we can install provider specific chartconfigs.

We need this for ingress-controller as we'll roll this out on Azure first. Since we don't need to worry about upgrades yet.

@r7vme also needs to add an Azure specific chart foe external-dns as part of enabling services of type loadbalancer on Azure.

### CODE REVIEW

File: service/controller/aws/v4/resource_set.go [hunk 1]
```diff
@@ -26,6 +26,11 @@ import (
 	"github.com/giantswarm/cluster-operator/service/controller/aws/v4/resource/awsconfig"
 )
 
+const (
+	// Provider is used to enable provider specific features.
+	Provider = "aws"
```
Comment (rossf7) [inline service/controller/aws/v4/resource_set.go]: Provider is set via constants in the resource set. This works I'm open to a more elegant solution if there is one.

File: service/controller/azure/cluster.go [hunk 2]
```diff
@@ -18,6 +18,7 @@ import (
 	"github.com/giantswarm/cluster-operator/service/controller/azure/v1"
 	"github.com/giantswarm/cluster-operator/service/controller/azure/v2"
 	"github.com/giantswarm/cluster-operator/service/controller/azure/v3"
+	"github.com/giantswarm/cluster-operator/service/controller/azure/v4"
```
Comment (rossf7) [inline service/controller/azure/cluster.go]: v4 is wired up for Azure.

File: pkg/v4/resource/chartconfig/create_test.go [hunk 3]
```diff
@@ -183,6 +183,7 @@ func Test_ChartConfig_newCreateChange(t *testing.T) {
 		K8sClient:   clientgofake.NewSimpleClientset(),
 		Logger:      microloggertest.New(),
 		ProjectName: "cluster-operator",
+		Provider:    "aws",
```
Comment (tuommaki) [inline pkg/v4/resource/chartconfig/create_test.go]: In another PR I created `ProviderXYZ` labels in `label` package: https://github.com/giantswarm/cluster-operator/pull/132/files#diff-5e9724242fffab7ce692f8161cb209bc

Maybe take that and use here?

File: service/controller/aws/v4/resource_set.go (hunk 1 above)
Comment (tuommaki) [reply service/controller/aws/v4/resource_set.go]: As I noted above, I need to do similar thing for `certconfigs` resource and I created entries to `label` package.

**Comment by rossf7** [comment]
@r7vme v4 is WIP so you can add your changes there. You can base it on one of the current chartconfigs. It just needs the check for the Azure provider.

https://github.com/giantswarm/cluster-operator/blob/b324be33c26879cce395658c55f3e7b5c9aa21a0/pkg/v4/resource/chartconfig/desired.go#L37-L40

I'll merge this once CI is working again. But if you're blocked let me know and I'll merge sooner.

**Comment by r7vme** [comment]
all good, not blocked, thanks

File: pkg/v4/resource/chartconfig/delete_test.go [hunk 4]
```diff
@@ -380,6 +382,7 @@ func Test_ChartConfig_newDeleteChangeForUpdatePatch(t *testing.T) {
 		K8sClient:   clientgofake.NewSimpleClientset(),
 		Logger:      microloggertest.New(),
 		ProjectName: "cluster-operator",
+		Provider:    "aws",
```
Comment (tuommaki) [inline pkg/v4/resource/chartconfig/delete_test.go]: Missing `label.ProviderAWS`?

Comment (rossf7) [reply pkg/v4/resource/chartconfig/delete_test.go]: Thanks! Yes I'd missed that. Done.

File: integration/test/basic/project.yaml [hunk 5]
```diff
@@ -7,7 +7,7 @@ test:
     - "REGISTRY_PULL_SECRET=${REGISTRY_PULL_SECRET}"
     - "CLUSTER_NAME=${CLUSTER_NAME}"
     - "VAULT_TOKEN=mytoken"
-    - "COMMON_DOMAIN_GUEST=${COMMON_DOMAIN_GUEST}"
+    - "COMMON_DOMAIN=${COMMON_DOMAIN}"
```
Comment (rossf7) [inline integration/test/basic/project.yaml]: I've vendored e2e-harness and updated the env var to align with aws-operator. Hopefully this will fix e2e.

Comment (fgimenez) [reply integration/test/basic/project.yaml]: Ok, I've already added `COMMON_DOMAIN` to circleci env, not sure if changing the name of the awsconfig resource chart to apiextensions is also required https://github.com/giantswarm/cluster-operator/commit/b34b0fa8a3c4ec6e651d1a1ea1f794c28d9ddffd. The build is running, let's see.

Comment (rossf7) [reply integration/test/basic/project.yaml]: Thanks Fede!

I'd missed using the chart from apiextensions. I've added here too. I'll run e2e once we know how your build went.

Comment (fgimenez) [reply integration/test/basic/project.yaml]: It was still failing, turns out that aws-operator was complaining about a missing `credential-default` secret:
```
```

Comment (fgimenez) [reply integration/test/basic/project.yaml]: Master is green, merging it here should fix tests.

Comment (kopiczko) [reply integration/test/basic/project.yaml]: Sorry for the mess introduced. I missed e2e-harness upgrade here.
Thanks for sorting <3

Comment (rossf7) [reply integration/test/basic/project.yaml]: @fgimenez Yes thanks for sorting! <3 I'll rebase and re-run e2e.

**Comment by rossf7** [comment]
CI green. Merging.

**Pull request merged**